packages/azure-kusto-data/src/cloudSettings.ts

```diff
@@ -68,22 +68,19 @@ class CloudSettings {
                 // Axios library has a bug in browser, not propagating the status code, see: https://github.com/axios/axios/issues/5330
                 if ((isNodeLike && ex.response?.status === 404) || (!isNodeLike && (!ex.code || ex.code === AXIOS_ERR_NETWORK))) {
                     // For now as long not all proxies implement the metadata endpoint, if no endpoint exists return public cloud data
-                    this.cloudCache[kustoUri] = this.defaultCloudInfo;
+                    this.cloudCache[cacheKey] = this.defaultCloudInfo;
                 } else {
                     throw new Error(`Failed to get cloud info for cluster ${kustoUri} - ${ex}`);
                 }
             }
         }
-        return this.cloudCache[kustoUri];
+        return this.cloudCache[cacheKey];
     }
 
-    private normalizeUrl(kustoUri: string) {
+    private getCacheKey(kustoUri: string): string {
         const url = new URL(kustoUri);
-        const urlString = url.toString();
-        if (urlString.endsWith("/")) {
-            return urlString.slice(0, urlString.length - 1);
-        }
-        return urlString;
+        // Return only the protocol and host (includes port if non-standard)
+        return `${url.protocol}//${url.host}`;
```

Comment:
IIRC you need the path as well.
This is cause in cases of proxies the path is pointing to the cluster.

What's this change for?
If you are trying to get rid of the endpoint part (/Vx/rest/ct.), you should make sure to call this with the root URI and not the composed endpoint.

Reply:
Checked with Alon Adam and se la vi.